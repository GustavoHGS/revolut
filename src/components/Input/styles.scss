@import "assets/_variables.scss";

.input-container {
  display: flex;
  flex-direction: column;
  margin: 8px;
  min-width: 5em;
}

.input-label {
  font-family: GloberBold;
  color: $black-cool;
  font-size: 14px;
}

.input {
  border-radius: 3px;
  border: 1px solid $gray-300;
  outline: none;
  height: 2em;
  padding: 6px;
  margin-top: 6px;
  font-family: GloberRegular;
}

.input:focus {
  border: 2px solid $primary;
  box-shadow: 0 2px 2px 2px rgba(69, 65, 78, 0.08);
}
