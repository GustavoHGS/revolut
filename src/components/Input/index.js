/* eslint react/forbid-prop-types: off */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './styles.scss'

export default class Input extends Component {
  state = {}

  handleChange = (e) => {
    const { onChange } = this.props
    onChange(this.props.formatter(e.target.value))
  }

  render() {
    const {
      type, className, label, prefix, value, labelStyle,
    } = this.props
    return (
      <div className="input-container">
        <span className="input-label" style={labelStyle}>{label}</span>
        <div className="row" style={{ alignItems: 'center' }}>
          {prefix()}
          <input
            onChange={this.handleChange}
            value={value}
            className={classNames('input', className)}
            type={type}
            placeholder={this.props.placeholder}
            maxLength={this.props.maxLength}
          />
        </div>
      </div>
    )
  }
}

Input.propTypes = {
  onChange: PropTypes.func,
  type: PropTypes.string,
  className: PropTypes.string,
  label: PropTypes.string,
  prefix: PropTypes.func,
  value: PropTypes.any,
  placeholder: PropTypes.any,
  formatter: PropTypes.func,
  labelStyle: PropTypes.object,
  maxLength: PropTypes.any,
}

Input.defaultProps = {
  onChange: () => {},
  type: 'text',
  className: '',
  label: '',
  prefix: () => {},
  value: '',
  placeholder: '',
  formatter: null,
  labelStyle: null,
  maxLength: null,
}
