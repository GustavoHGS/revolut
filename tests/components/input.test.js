import React from 'react'
import renderer from 'react-test-renderer'
import Input from '../../src/components/Input'

describe('Input Component', () => {
  it ('matches the input snapshot', () => {
    const nodeTree = renderer.create(<Input />).toJSON()
    expect(nodeTree).toMatchSnapshot()
  })
})
