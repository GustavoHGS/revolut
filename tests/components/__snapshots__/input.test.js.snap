// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches the input snapshot 1`] = `
<div
  className="input-container"
>
  <span
    className="input-label"
    style={null}
  >
    
  </span>
  <div
    className="row"
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <input
      className="input"
      maxLength={null}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </div>
</div>
`;
